graphprot/NeuralNet.py

```diff
@@ -17,51 +17,54 @@
 from .DataSet import HDF5DataSet, DivideDataSet, PreCluster
 from .Metrics import Metrics
 
+
 class NeuralNet(object):
 
     def __init__(self, database, Net,
                  node_feature=['type', 'polarity', 'bsa'],
                  edge_feature=['dist'], target='irmsd',
                  batch_size=32, percent=[0.8, 0.2], index=None, database_eval=None,
-                 class_weights=None, task='class', classes=[0, 1], threshold=4):
+                 class_weights=None, task='class', classes=[0, 1], threshold=4, pretrained_model=None):
+
+        # load the input data or a pretrained model
+        if pretrained_model is None:
+            for k, v in dict(locals()).items():
+                if k not in ['self', 'database', 'Net', 'database_eval']:
+                    self.__setattr__(k, v)
+
+        else:
+            self.load_params(pretrained_model)
 
         # dataset
-        dataset = HDF5DataSet(root='./', database=database, index=index,
-                              node_feature=node_feature, edge_feature=edge_feature,
-                              target=target)
-        #PreCluster(dataset, method='mcl')
+        dataset = HDF5DataSet(root='./', database=database, index=self.index,
+                              node_feature=self.node_feature, edge_feature=self.edge_feature,
+                              target=self.target)
+        # PreCluster(dataset, method='mcl')
 
+        # divide the dataset
         train_dataset, valid_dataset = DivideDataSet(
-            dataset, percent=percent)
+            dataset, percent=self.percent)
+
+        # dataloader
+        self.train_loader = DataLoader(
+            train_dataset, batch_size=self.batch_size, shuffle=False)
 
```

Comment:
Shouldn't we shuffle the data automatically ? or add it as an option ?

Reply:
I think we should. I thought you wrote that but it might be me :) I'll add it as an option